tcpsock.go

```diff
@@ -119,6 +120,35 @@ func ResolveTCPAddr(network, address string) (*TCPAddr, error) {
 	return &TCPAddr{IP: ip.AsSlice(), Port: port}, nil
 }
 
+// ResolveIPAddr returns an address of IP end point.
+//
+// The network must be an IP network name.
+//
+// If the host in the address parameter is not a literal IP address,
+// ResolveIPAddr resolves the address to an address of IP end point.
+// Otherwise, it parses the address as a literal IP address.
+// The address parameter can use a host name, but this is not
+// recommended, because it will return at most one of the host name's
+// IP addresses.
+//
+// See func [Dial] for a description of the network and address
+// parameters.
+func ResolveIPAddr(network, address string) (*IPAddr, error) {
```

Comment:
Move this func to iprawsock.go please.  I'm trying to keep things aligned with where things are in the upstream src/net repo, for comparison sake.